### PR TITLE
Fix collision between flake8 and yapf

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Related: https://github.com/PyTorchLightning/pytorch-lightning/pull/5591

### Example of collision
Currently, applying yapf can cause flake8's error `[W503] line break before binary operator` which was found in https://github.com/PyTorchLightning/pytorch-lightning-bolts/pull/462/files#r561509665.

### Description of the changes
- Ignore both `W503` and `W504`. (If we want to keep the codebase compliant with PEP8 regarding this, then we should only ignore `W503`. [related pep8](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator))
- Ignore `E129`

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
I'm getting very confused, so let me leave a note here.

### Line break occurred before a binary operator (W503)
```python
# bad
income = (gross_wages
          + taxable_interest)
# good
income = (gross_wages +
          taxable_interest)
```

### Line break occurred after a binary operator (W504)
```python
# bad
income = (gross_wages +
          taxable_interest)
# good
income = (gross_wages
          + taxable_interest)
```

**Comment by akihironitta** [comment]
Not sure why `yapf` is happy about the following lines where `split_before_arithmetic_operator = true` should be applied:
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/ea59b64d4fa743337fbc770389339cb3946eaa34/pl_bolts/optimizers/lr_scheduler.py#L117-L118

**Comment by akihironitta** [comment]
yapf also conflict with `E129` from flake8... Let me look into it...

File: setup.cfg [hunk 1]
```diff
@@ -32,15 +32,19 @@ verbose = 2
...
+    W504  # Ignore "Line break occurred after a binary operator"
+    W605  # Ignore "Invalid escape sequence 'x'"
 
 [yapf]
 based_on_style = pep8
 spaces_before_comment = 2
 split_before_logical_operator = true
+split_before_arithmetic_operator = true
+split_before_bitwise_operator = true
 COLUMN_LIMIT = 120
```
Comment (akihironitta) [inline setup.cfg:48]: If we ignore both `W503` and `W504`, we can just remove these additions.
```suggestion
split_before_logical_operator = true
COLUMN_LIMIT = 120
```

File: pl_bolts/models/self_supervised/simclr/simclr_module.py [hunk 2]
```diff
@@ -38,8 +38,8 @@ def backward(ctx, grad_output):
         grad_input = grad_output.clone()
         torch.distributed.all_reduce(grad_input, op=torch.distributed.ReduceOp.SUM, async_op=False)
 
-        return grad_input[torch.distributed.get_rank() * ctx.batch_size:(torch.distributed.get_rank() + 1) *
-                          ctx.batch_size]
+        return grad_input[torch.distributed.get_rank() * ctx.batch_size:(torch.distributed.get_rank() + 1)
+                          * ctx.batch_size]
```
Comment (Borda) [inline pl_bolts/models/self_supervised/simclr/simclr_module.py:42]:
```suggestion
        idx_from = torch.distributed.get_rank() * ctx.batch_size
        idx_to = (torch.distributed.get_rank() + 1) * ctx.batch_size
        return grad_input[idx_from:idx_to]
```

File: setup.cfg [hunk 3]
```diff
@@ -32,15 +32,19 @@ verbose = 2
...
-    E731  # Do not assign a lambda expression, use a def
-    E231  # Ignore missing space after comma
-    W504  # Line break occurred after a binary operator
-    W605  # Invalid escape sequence 'x'
+    E129  # Ignore "Visually indented line with same indent as next logical line"
+    E231  # Ignore "Missing whitespace after ',', ';', or ':'"
+    E731  # Ignore "Do not assign a lambda expression, use a def"
+    W503  # Ignore "Line break occurred before a binary operator"
+    W504  # Ignore "Line break occurred after a binary operator"
+    W605  # Ignore "Invalid escape sequence 'x'"
```
Comment (Borda) [inline setup.cfg:40]:
```suggestion
    E231  # Ignore "Missing whitespace after ',', ';', or ':'"
    E731  # Ignore "Do not assign a lambda expression, use a def"
    W503  # Ignore "Line break occurred before a binary operator"
    W504  # Ignore "Line break occurred after a binary operator"
    W605  # Ignore "Invalid escape sequence 'x'"
```

File: pl_bolts/datamodules/async_dataloader.py [hunk 4]
```diff
@@ -92,10 +92,9 @@ def __iter__(self) -> "AsynchronousLoader":
         # We don't want to run the thread more than once
         # Start a new thread if we are at the beginning of a new epoch, and our current worker is dead
 
-        # yapf: disable
-        if (not hasattr(self, 'worker') or not self.worker.is_alive()) and self.queue.empty() and self.idx == 0:  # type: ignore[has-type] # noqa: E501
+        if (not hasattr(self, 'worker')
+            or not self.worker.is_alive()) and self.queue.empty() and self.idx == 0:  # type: ignore[has-type]
```
Comment (Borda) [inline pl_bolts/datamodules/async_dataloader.py:96]:
```suggestion
        if_worker = (not hasattr(self, 'worker') or not self.worker.is_alive())
        if if_worker and self.queue.empty() and self.idx == 0:  # type: ignore[has-type]
```

**Comment by Borda** [comment]
seem to be a kind of deadlock, so I would recommend simplifying the statements...

**Pull request merged**